components/remote_settings/src/service.rs

```diff
@@ -81,13 +85,30 @@ impl RemoteSettingsService {
         // Make sure we only sync each collection once, even if there are multiple clients
         let mut synced_collections = HashSet::new();
 
-        // TODO: poll the server using `/buckets/monitor/collections/changes/changeset` to fetch
-        // the current timestamp for all collections.  That way we can avoid fetching collections
-        // we know haven't changed and also pass the `?_expected{ts}` param to the server.
+        let mut inner = self.inner.lock();
+        let changes = inner.fetch_changes()?;
+        let change_map: HashMap<_, _> = changes
+            .changes
+            .iter()
+            .map(|c| ((c.collection.as_str(), &c.bucket), c.last_modified))
+            .collect();
+        let bucket_name = inner.bucket_name.clone();
 
-        for client in self.inner.lock().active_clients() {
-            if synced_collections.insert(client.collection_name()) {
-                client.internal.sync()?;
+        for client in inner.active_clients() {
+            let client = &client.internal;
+            let collection_name = client.collection_name();
+            if let Some(client_last_modified) = client.get_last_modified_timestamp()? {
+                if let Some(server_last_modified) = change_map.get(&(collection_name, &bucket_name))
+                {
+                    if client_last_modified >= *server_last_modified {
```

Comment:
Here I would advise to use `!=` instead. This way, if the local data is tempered a very high timestamp is set, the sync will not be skipped